alma_sbom.py

```diff
@@ -49,6 +32,59 @@ class PackageNevra:
     release: str = None
     arch: str = None
 
+    def __repr__(self):
+        if self.epoch is not None:
+            return f'{self.epoch}:{self.name}-' \
+                   f'{self.version}-{self.release}.{self.arch}'
+        else:
+            return f'{self.name}-{self.version}-' \
+                   f'{self.release}.{self.arch}'
+
+
+@dataclasses.dataclass
+class FileFormat:
+    sbom_record_type: Literal[
+        'cyclonedx',
+        'spdx',
+    ] = 'cyclonedx'
+    file_format: Literal[
+        'json',
+        'xml'
+    ] = 'json'
+
+    def __repr__(self):
+        return f'"{self.sbom_record_type},{self.file_format}"'
+
+
+class FileFormatType(object):
+
+    supported_file_formats = defaultdict(list, **{
+        'cyclonedx': [
+            'json',
+            'xml',
+        ]
+    })
+
+    def __call__(self, sbom_type_file_format: str) -> FileFormat:
+        sbom_record_type, file_format = sbom_type_file_format.split(',')
```

Comment:
Sorry, I overlooked this. If memory serves me well (can't find the notes from that meeting), Eugene suggested to receive "cyclonedx-xml" or "cyclonedx-json", anything against doing it?

Reply:
Nothing, done